src/main/java/org/testng/internal/ClassImpl.java

```diff
@@ -123,6 +123,7 @@ private Object getInstanceFromGuice() {
     return injector.getInstance(m_class);
   }
 
+  @SuppressWarnings("unchecked")
```

Comment:
Can we replace `Suite suite.getParentModule()` by `Class<Model> suite.getParentModule()` instead?

Reply:
`suite.getParentModule()` currently returns a String which is being instantiated via reflection here. So we cannot do like what you have suggested without adding that new method within the `XmlSuite` object. And even if we add the new method to `XmlSuite` object we will still be stuck with the same problem because we only get a Fully Qualified Classname from the suite xml file, and have to mandatorily rely on reflection to load the class, and when we do that we would need to do a `unchecked` class casting. So I am of the opinion that we leave this as is, instead of introducing more changes

Reply:
Ok no prob but in order to clarify my point of view: `XmlSuite` is the xml model and host the string value and Suite is the object view and should host Class.

Reply:
Got it, Oh btw, please let me know if there are any other observations that need to be addressed or if this PR is good to be approved.